fossology/fossology
# feat(ununpack): Creating logs if an unpack attempt fails

## Enhancements
1.  Preserve the unpack tool output for log purpose.
1.  Use popen() instead of system() to capture program output.
1.  Print the output to log if tool's return code is `NOT OK` or if verbose mode is on.

## Changes
*  **Departition:** Put few messages in verbose only preventing unwanted output.
*  **ununpack.c**: Change the alignments.

## How to test
1.  Upload a normal file with verbose off.
    -  No log should be generated.
1.  Upload a normal file with verbose on.
    -  Log should contain information form tool.
1.  Upload a dirty package with verbose off.
    -  Log should contain the failure message along with return code.
1.  Upload a dirty package with verbose on.
    -  Log should contain the failure message along with return code.

>  P.S. Log mentioned is the one which comes in **Geeky Scan Details**.
